File: python/singleNumber.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        nums.sort()
        lone = [nums[0]]
        for idx in range(1, len(nums)):
            if len(lone) == 0:
                lone.append(nums[idx])
            elif lone[-1] != nums[idx]:
                lone.append(nums[idx])
            else:
                lone.pop()
        return lone[0]

File: cpp/singleNumber.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        unordered_map<int,int> num_to_freq;
        unordered_map<int,int> freq_to_num;
        for(int num:nums){
            if(num_to_freq.find(num)==num_to_freq.end())
                num_to_freq[num] = 1;
            else
                num_to_freq[num]++;
        }

        vector<int> ret;
        for (int num:nums){
            if(num_to_freq[num]==1){
               ret.push_back(num);
            }
        }

        return ret;
    }
};